cEP-0020.md

```diff
@@ -0,0 +1,244 @@
+
+As coala tries to be a welcoming organization for newcomers, it gives them
+clear pathways and as much direct assistance as possible. However, it is
+already known that newcomers face many barriers while attempting to contribute
+to open source for the first time, and additional barriers while attempting
+to contribute to projects run by the coala organisation.
+
+Some barriers they face include orientation issues that can potentially
+demotivate newcomers from placing their first contribution. On the other
+hand, gamification is widely used to engage and motivate people to
+accomplish tasks and improve their performance.
+
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Gamification Elements
+
+There will be four game design elements to help newcomers: Activities, Points,
+Levels and Badges. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create Issues
+  - Labels?
+  - No. of comments?
+- Create Pull Requests
+  - Label?
+  - CI status?
+  - No. of Gitmate errors?
+  - No. and type of Comments?
+- Comments
+  - Type?
+- Gitter Messages
+  - Good?
+  - Bad?
+
+The above activities will be tracked seperately based on their category. i.e.:
+
+- There are five types of Issues and Pull Requests that could be tracked:
+  - type/bug
+  - difficulty/newcomer
+  - difficulty/low
+  - difficulty/medium
+  - difficulty/high
+- While tracking Pull Requests, following factors will also be considered:
+  - CI status
+    - Passed
+    - Failed
+  - Comments
+    - No. of simple comments
+    - No. of review comments
+  - Gitmate Errors:
+    - 1 to 5
+    - 5 to 10
+    - 10 and beyond
+- Following are the types of comments that could be tracked:
+  - Issue comments
+  - Review comments
+
+### Points
+
+Points are a simple tool to reward behavior, and they provide an excellent
+base upon which to build a gamified design. Now lets assign some points
+to each of the tracked activities mentioned above:
+
+- Create Issue
+  - 10 points for difficulty/newcomer
+  - Increase in 10 points based on difficulty labels
+  - Increase in 2 points with the increase in no. of comments
+- Create Pull Requests
+  - If CI passes and There are no Gitmate errors
+    - 15 points for intial and then increase
+  - If CI passes and Gitmate errors are b/w 1 to 5
+    - 10 points for intial and then increase
+  - If CI passes and Gitmate errors are b/w 5 to 10
+    - 5 points for intial and then increase
+  - If CI passes and Gitmate errors are b/w 10 and beyond
+    - No points given
+  - If CI failed
+    - No points given until the CI has passed
+- Made Comments
+  - 2 points per issue comments
+  - 3 points per review comments
+- Gitter Messages
+  - No points for Bad messages
```

Comment:
we need negative points.  that is one of the primary goals of this project from a maintainer perspective -- we need to be able to describe & strongly discourage abusive behaviour, where 'abusive' means 'wastes maintainer time'.